### ISSUE TITLE
[immutability] Detect mutations on for-of iterator variables from props/state

### ISSUE THREAD

**Issue by Rel1cx** [Rel1cx/eslint-react#1764]
## Background
Per `immutability.spec.diff.md`, the current `immutability` rule does not detect mutations on `for-of` iterator variables, even when the iterated collection comes from props or state.

## Current behavior
```jsx
function Component({ items }) {
  for (const x of items) {
    x.modified = true; // ❌ currently missed
  }
}
```

## Expected behavior
When the right-hand side of a `for-of` (e.g. `items`) is identified as coming from props or state, treat the left-hand iterator variable (e.g. `x`) as an immutable object and flag mutations on it.

## References
- Diff report: `plugins/eslint-plugin-react-x/src/rules/immutability/immutability.spec.diff.md` (§8 / §9 Key Gaps #7)
- SPEC: `ValidateNoFreezingKnownMutableFunctions`

## Priority
P2 — medium implementation cost, covers a common indirect mutation pattern.
